services/git-service-api/src/main/java/io/fabric8/launcher/service/git/api/GitRepositoryFilter.java

```diff
@@ -0,0 +1,19 @@
+package io.fabric8.launcher.service.git.api;
+
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable(singleton = true)
+@JsonSerialize(as = ImmutableGitRepositoryFilter.class)
+@JsonDeserialize(as = ImmutableGitRepositoryFilter.class)
+public interface GitRepositoryFilter {
+
+    @Nullable String withNameContaining();
+
+    @Value.Parameter
+    @Nullable GitOrganization withOrganization();
```

Comment:
I think it would be easier to make them return `Optional` objects to avoid the boilerplate of "if (.. == null)", but that's fine

Reply:
We could talk forever about this one :)
At the end, I think, the most important is to have a common approach on how we use `Optional`. I don't mind if you want me to change it..

Reply:
Not asking to Change it, just an idea ;)